### PR TITLE
refactor: test_context.go

### PR DESCRIPTION
**Pull request by longyue0521**
### 🤔 What's changed?

1. ScenarioContext has methods on both value and pointer receivers, make all methods have value receivers.
2. fix the syntax error in the comment
3. remove unused function Build

### ⚡️ What's your motivation? 

make the code more cleanable, readable, more compliant with Go language specifications

### 🏷️ What kind of change is this?

- [x] :bank: Refactoring/debt/DX (improvement to code design, tooling, documentation etc. without changing behaviour)
- :bug: Bug fix (non-breaking change which fixes a defect)
- :zap: New feature (non-breaking change which adds new behaviour)
- :boom: Breaking change (incompatible changes to the API)

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [ ] I've changed the behaviour of the code
  - [ ] I have added/updated tests to cover my changes.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] Users should know about my change
  - [ ] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

**Comment by longyue0521** [comment]
@vearutop

File: test_context.go [hunk 1]
```diff
@@ -331,20 +330,4 @@ func (ctx *ScenarioContext) stepWithKeyword(expr interface{}, stepFunc interface
...
-//
-// If there are go test files, it first builds a test
-// package with standard go test command.
-//
-// Finally it generates godog suite executable which
-// registers exported godog contexts from the test files
-// of tested package.
-//
-// Returns the path to generated executable
-func Build(bin string) error {
```
Comment (vearutop) [inline test_context.go:346]: Removing this function would be a breaking change, let's deprecate it instead.

**Comment by vearutop** [comment]
Hi, thank you, this change seems like a safe simplification. Please restore the `Build` function and I think it'll be good to merge.

File: test_context.go (hunk 1 above)
Comment (longyue0521) [reply test_context.go:346]: when add `// Deprecated: use xxxx instead` we should tell user which function to instead ?

**Comment by longyue0521** [comment]
> Hi, thank you, this change seems like a safe simplification. Please restore the `Build` function and I think it'll be good to merge.

sorry, I don't know which function to instead the `Build` function, I just rollback it.

**Review by vearutop — APPROVED** [review]

File: test_context.go (hunk 1 above)
Comment (vearutop) [reply test_context.go]: You can use something free formed like `// Deprecated: feature will be removed, no substitute`. The message comes from your reasoning why you wanted to remove that function in the first place.

It's ok to keep it as is for now.

**Review by aviscode — APPROVED** [review]

**Comment by aslakhellesoy** [comment]
Hi @longyue0521,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Pull request merged**